cpp/inc/bond/protocol/simple_json_reader.h

```diff
@@ -35,26 +36,27 @@ class SimpleJsonReader
     BOND_STATIC_CONSTEXPR uint16_t version = 0x0001;
 
     /// @brief Construct from input buffer/stream containing serialized data.
-    SimpleJsonReader(typename boost::call_traits<Buffer>::param_type input)
-        : _input(input),
-          _stream(_input),
+    SimpleJsonReader(const Buffer& input)
+        : _streamHolder(input),
           _document(boost::make_shared<rapidjson::Document>()),
-          _value(NULL)
-    {}
-
-    SimpleJsonReader(const SimpleJsonReader& that, const Field& value)
-        : _stream(that._stream),
-          _document(that._document),
+          _value(nullptr)
+    { }
+
+    /// @brief Create a "child" SimpleJsonReader to read \c value, which is
+    /// known to be a member of the document in \c parent.
+    ///
+    /// @warning \c parent must remain alive for the lifetime of this child.
+    SimpleJsonReader(SimpleJsonReader& parent, const Field& value)
+        : _streamHolder(parent),
+          _document(parent._document),
           _value(&value)
-    {}
+    {
+        // Must have an already-parsed parent
+        BOOST_ASSERT(parent._value);
+    }
 
     /// @brief Copy constructor
-    SimpleJsonReader(const SimpleJsonReader& that)
-        : _input(that._input),
-          _stream(that._stream, _input),
-          _document(that._document),
-          _value(that._value)
-    {}
+    SimpleJsonReader(const SimpleJsonReader&) = default;
```

Comment:
I think this is not needed.

Reply:
It was just there for Doxygen. The library is inconsistent in documenting what is and isn't copyable.
I've, instead, encoded that a `SimpleJsonReader` is copyable (for `InputBuffer` at least) in the json_tests unit test with `static_assert` tests.